src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3041,10 +3072,20 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
             startingState.ForEach(
```

Comment:
>startingState [](http://example.com/codeflow?start=12&length=13)

The fact that we are using ```startingState``` as a template when we are visiting an "unreachable" local function for the first time feels very suspicious. Why do we care what was the starting state? If it already has all the right slots with right values, then we should simply use it. If it doesn't have the right things, then why use it as a template?